uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/leave-request-popup-controller_test.js

```diff
@@ -554,6 +569,147 @@
           });
         });
       });
+
+      describe('when absence period is changed', function () {
+        describe('for multiple days', function () {
+          describe('before from date is selected', function () {
+            it('should disable to date and to type', function () {
+              expect($ctrl.leaveRequest.from_date).toBeFalsy();
+            });
+          });
+
+          describe('after from date is selected', function () {
+            beforeEach(function () {
+              setTestDates(date2017);
+            });
+
+            it('should enable to date and to type', function () {
+              expect($ctrl.leaveRequest.from_date).toBeTruthy();
+            });
+
+            it('should check if date is in any absence period', function () {
+              expect($ctrl.error).toBeNull();
+            });
+
+            it('should update calendar', function () {
+              expect(WorkPatternAPI.getCalendar).toHaveBeenCalled();
+            });
+
+            it('should not show balance', function () {
+              expect($ctrl.uiOptions.showBalance).toBeFalsy();
+            });
+
+            describe('from available absence period', function () {
+              var oldPeriodId;
+
+              beforeEach(function () {
+                $ctrl.uiOptions.toDate = null;
+                oldPeriodId = $ctrl.period.id;
+                setTestDates(date2016);
+              });
+
+              it('should change absence period', function () {
+                expect($ctrl.period.id).not.toEqual(oldPeriodId);
+              });
+
+              it('should set min and max to date', function () {
+                expect($ctrl.uiOptions.date.to.options.minDate).not.toBeNull();
+                expect($ctrl.uiOptions.date.to.options.maxDate).not.toBeNull();
+              });
+
+              it('should update absence types from Entitlements', function () {
+                expect(EntitlementAPI.all).toHaveBeenCalled();
+              });
+
+              it('should not show balance', function () {
+                expect($ctrl.uiOptions.showBalance).toBeFalsy();
+              });
+
+              it('should reset to date', function () {
+                expect($ctrl.leaveRequest.to_date).toBeNull();
+              });
+            });
+
+            describe('from unavailable absence period', function () {
+              beforeEach(function () {
+                setTestDates(date2013);
+              });
+
+              it('should show error', function () {
+                expect($ctrl.error).toEqual(jasmine.any(String));
+              });
+            });
+
+            describe('and to date is selected', function () {
+              beforeEach(function () {
+                setTestDates(date2016, date2016);
+              });
+
+              it('should select date from selected absence period', function () {
```

Comment:
Title and expect does not match, maybe a better title
`should select date from selected absence period without errors`